src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -138,4 +140,25 @@ public function validate($entity, Constraint $constraint)
             ->setCode(UniqueEntity::NOT_UNIQUE_ERROR)
             ->addViolation();
     }
+
+    /**
+     * @param ObjectManager $em
+     * @param ClassMetadata $class
+     * @param object        $entity
+     *
+     * @return string
+     */
+    private function buildInvalidValueString(ObjectManager $em, ClassMetadata $class, $entity)
+    {
+        $identifiers = array_map(function ($identifier) use ($em) {
+            // identifiers can be objects (without any __toString method) if its a composite PK
+            if (is_object($identifier) && !method_exists($identifier, '__toString')) {
+                return sprintf('(%s)', $this->buildInvalidValueString($em, $em->getClassMetadata(get_class($identifier)), $identifier));
```

Comment:
is it safe to use the same object manager here for the related class? Not sure if Doctrine supports relations between Classes of different object managers?